Standardise examples

asherbender opened the pull request:
Resolves #8234

This pull requests standardises the simple crop simulations in Toowoomba (that use the `Dalby.met` weather file), including:

- [Barley](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/Examples/Barley.apsimx)
- [Maize](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/Examples/Maize.apsimx)
- [Mungbean](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/Examples/Mungbean.apsimx)
- [Oats](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/Examples/Oats.apsimx)
- [Soybean](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/Examples/Soybean.apsimx)
- [Wheat](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/Examples/Wheat.apsimx)

These simulations broadly followed the same structure. This pull request standardises the structure to be:

```
+- Simulations
    +- Simulation
    |   +- Clock
    |   +- Summary
    |   +- Weather
    |   +- SoilArbitrator
    |   +- Field
    |       +- Report
    |       +- Fertiliser
    |       +- Soil
    |       |   +- Physical
    |       |   |   +- <crop>Soil
    |       |   +- SoilWater
    |       |   +- Organic
    |       |   +- Chemical
    |       |   +- Water
    |       |   +- Temperature
    |       |   +- Nutrient
    |       |   +- NO3
    |       |   +- NH4
    |       |   +- Urea
    |       +- SurfaceOrganicMatter
    |       +- MicroClimate
    |       +- <crop>
    |       +- <sow>
    |       +- <fertilise>
    |       +- <harvest>
    |       +- Graph
    |           +- Series
    +- DataStore
```

This provides a consistency in model names and order across the included examples.

Additionally:

- All standardised examples use the ApSoil profile: `Soils/Australia/Queensland/Darling Downs and Granite Belt/Black Vertosol-Mywybilla (Norwin No900)`.
    - Note that the values in the ApSoil profile are *identical* to the original profile used in the simulations. The change only causes minor difference in the `Soil` model metadata which now mirror the ApSoil profile.

- All examples are updated so that `Report` outputs yield in `kg/ha`.
    - For some simulations, the line `[<crop>].Grain.Total.Wt*10 as Yield` was added

- The management actions (sowing, fertilising and harvesting) all use the most up-to-date scripts from the [ManagementToolbox](https://github.com/APSIMInitiative/ApsimX/blob/0fce476e9b16433dd2b1fa40047174cf04490c94/ApsimNG/Resources/Toolboxes/ManagementToolbox.apsimx).

- The standardised simulations output identical values to the old simulations. In cases where model names were changed or extra `Report` output was added, the output will differ in these minor ways.

hut104 commented:
It is great to see these being cleaned up.  Just two suggestions
1) We should standardise example to have Microclimate in the Top level (ie next to soil arbitrator).  It can operate at the paddock level, but that won't work for multi-point simulations.  I suggest that we change microclimate to only allow a position in the top level.
2) Can we remove the aliasing of yield to kg/ha, and simply output wheat.grain.wt etc.  Then remove the graph y-axis title value of "yield" so that the graph simply gets the name from the output variable name.  If people copy the graph, it will retain the y-axis title of "yield" even if they change the reporting variable for the graph series.  I notice lots of people with graphs that have the Y-axis title of yield because they copied these simulations and graphs within them.

hol353 commented:
add to whitelist jenkins

par456 commented:
retest this please jenkins

par456 commented:
retest this please jenkins

asherbender commented:
@hut104 

> It is great to see these being cleaned up. Just two suggestions
> 
> 1. We should standardise example to have Microclimate in the Top level (ie next to soil arbitrator).  It can operate at the paddock level, but that won't work for multi-point simulations.  I suggest that we change microclimate to only allow a position in the top level.

Happy to implement this change... Let me know if you would like me to update this pull request - or whether that requires changes to `microclimate` first.

> 2. Can we remove the aliasing of yield to kg/ha, and simply output wheat.grain.wt etc.  Then remove the graph y-axis title value of "yield" so that the graph simply gets the name from the output variable name.  If people copy the graph, it will retain the y-axis title of "yield" even if they change the reporting variable for the graph series.  I notice lots of people with graphs that have the Y-axis title of yield because they copied these simulations and graphs within them.

This is an easy change! Happy to update the files to plot `<crop>.grain.wt`. Again - confirm if you'd like me to update this pull request

Cheers!
-A

hol353 commented:
@asherbender : If you could update your PR as per @hut104 suggestions that would be great. Don't worry about changing Microclimate source code so that it only works at the top level. That's a bigger change for another PR.

asherbender commented:
@hut104, @hol353,

Sorry about the delay! The simulations now look like:

```
+- Simulations
    +- Simulation
    |   +- Clock
    |   +- Summary
    |   +- Weather
    |   +- SoilArbitrator
    |   +- MicroClimate
    |   +- Field
    |       +- Report
    |       +- Fertiliser
    |       +- Soil
    |       |   +- Physical
    |       |   |   +- <crop>Soil
    |       |   +- SoilWater
    |       |   +- Organic
    |       |   +- Chemical
    |       |   +- Water
    |       |   +- Temperature
    |       |   +- Nutrient
    |       |   +- NO3
    |       |   +- NH4
    |       |   +- Urea
    |       +- SurfaceOrganicMatter
    |       +- <crop>
    |       +- <sow>
    |       +- <fertilise>
    |       +- <harvest>
    |       +- Graph
    |           +- Series
    +- DataStore
```

1. `Microclimate` has been moved to below `SoilArbitrator`
2.  I have linked to `<crop>.Grain.Total.Wt` in the `Series` plots. The X-axis retains the label 'Date' but the Y-axis label is left blank (auto-configures). For the same reason, I removed the `Graph` title - it defaults to 'Graph'.

Although I changed the `Graph`, I left 'Yield' in the report - it demonstrates how `Report` variables can be manipulated and aliased. Parameter included in `Report` are largely left as they were in the original files. Do you want to implement a 'standard' here as well?

hol353 commented:
Well done. Merging this. I'll add another GitHub issue for enforcing Microclimate to run at the top level.
